### PR TITLE
[NumpyMatrixOperator] rename '_matrix' attribute to 'matrix'

### PR DESCRIPTION
**Pull request by sdrave**
Subclasses of `ImmutableInterface` should have equally-named attributes for their `__init__` attributes. Otherwise a custom `with_` implementation is needed. (Right now, `NumpyMatrixOperator._with` is broken.) Also, you often need access to the `matrix` attribute in interactive work, and tab completion will not show private attributes.

Any objections, @pymor/pymor-devs?

### CODE REVIEW

**Comment by ftalbrecht** [comment]
no objections

File: src/pymor/algorithms/to_matrix.py [hunk 1]
```diff
@@ -51,17 +51,17 @@ def __init__(self, format, mu):
...
             if not op.sparse:
-                return op._matrix
+                return op.matrix
             else:
-                return op._matrix.toarray()
+                return op.matrix.toarray()
         else:
             if not op.sparse:
-                return getattr(sps, format + '_matrix')(op._matrix)
+                return getattr(sps, format + 'matrix')(op.matrix)
```
Comment (pmli) [inline src/pymor/algorithms/to_matrix.py]: The underscore is necessary in `format + '_matrix'`, e.g. to use `csc_matrix`.

File: src/pymor/algorithms/to_matrix.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def action_AdjointOperator(self, op):
             else:
                 res = spsla.spsolve(source_product, res)
         if format is not None and format != 'dense':
-            res = getattr(sps, format + '_matrix')(res)
+            res = getattr(sps, format + 'matrix')(res)
```
Comment (pmli) [inline src/pymor/algorithms/to_matrix.py]: See above.

File: src/pymor/algorithms/to_matrix.py [hunk 3]
```diff
@@ -146,7 +146,7 @@ def action_VectorArrayOperator(self, op):
         format = self.format
         res = op._array.data if op.transposed else op._array.data.T
         if format is not None and format != 'dense':
-            res = getattr(sps, format + '_matrix')(res)
+            res = getattr(sps, format + 'matrix')(res)
```
Comment (pmli) [inline src/pymor/algorithms/to_matrix.py]: See above.

File: src/pymor/algorithms/to_matrix.py [hunk 4]
```diff
@@ -157,4 +157,4 @@ def action_ZeroOperator(self, op):
         elif format == 'dense':
             return np.zeros((op.range.dim, op.source.dim))
         else:
-            return getattr(sps, format + '_matrix')((op.range.dim, op.source.dim))
+            return getattr(sps, format + 'matrix')((op.range.dim, op.source.dim))
```
Comment (pmli) [inline src/pymor/algorithms/to_matrix.py]: See above.

**Pull request merged**